extensions/MasterMath/midi.js

```diff
@@ -0,0 +1,271 @@
+        } else if (event.port.state == "disconnected") {
+          // Find and remove from midiInputDevices
+          const inputIndex = midiInputDevices.findIndex(
+            (item) =>
+              item === `[id: "${event.port.id}" name: "${event.port.name}"]`
+          );
+          if (inputIndex !== -1) midiInputDevices.splice(inputIndex, 1);
+
+          // Find and remove from midiDeviceInfo
+          const infoIndex = midiDeviceInfo.findIndex(
+            (item) => item[0] === event.port.id
+          );
+          if (infoIndex !== -1) midiDeviceInfo.splice(infoIndex, 1);
+        }
+      };
+
+      function onMIDIMessage(event) {
+        const [status, note, velocity] = event.data;
+        const command = status & 0xf0;
+        if (command === 0x90 && velocity > 0) {
+          notesOn.push(note);
+          noteVelocities.push([note, velocity]);
+          lastNotePressed = note;
+          Scratch.vm.runtime.startHats("midi_whenAnyNote", {
+            pressedReleased: "pressed",
+          });
+          Scratch.vm.runtime.startHats("midi_whenNote", {
+            note: note,
+            pressedReleased: "pressed",
+          });
+        } else if (command === 0x80 || (command === 0x90 && velocity === 0)) {
+          lastNoteReleased = note;
+          notesOn.splice(notesOn.indexOf(note), 1);
+          noteVelocities.splice(
+            noteVelocities.findIndex((subArray) => subArray[0] === note),
+            1
+          );
+          Scratch.vm.runtime.startHats("midi_whenAnyNote", {
+            pressedReleased: "released",
+          });
+          Scratch.vm.runtime.startHats("midi_whenNote", {
+            note: note,
+            pressedReleased: "released",
+          });
+        } else {
+          console.log(
+            `Other MIDI Message: Status=${status}, Note=${note}, Velocity=${velocity}, Timestamp ${event.timeStamp}`
+          );
+        }
+      }
+
+      midiAccess.inputs.forEach((entry) => {
+        entry.onmidimessage = onMIDIMessage;
+      });
+    }
+
+    function onError(err) {
+      alert("MIDI Access Error:", err);
+      throw new Error("MIDI Access Error:", err);
+    }
+  } else {
+    alert("MIDI is not supported on this browser.");
+    throw new Error("MIDI is not supported on this browser.");
+  }
+
+  class MIDI {
+    getInfo() {
+      return {
+        id: "midi",
+        name: Scratch.translate("MIDI"),
+        blocks: [
+          {
+            opcode: "MIDIinputDevices",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("connected MIDI input devices"),
+            disableMonitor: true,
+          },
+          {
+            opcode: "midiDeviceInfo",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate("[info] of MIDI device [number]"),
+            arguments: {
+              info: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "name",
```

Comment:
Don't need `defaultValue` for any of the places where you defined `defaultValue` with a menu, since the menu options will already provide the one you want as the default as the first option based on how you made your list.